examples/rag_report_demo_why_some_customers_donot_like_pizza.sql

```diff
@@ -0,0 +1,187 @@
+-- Step 1. Preconfig & Check environment
+-- clean the history test table
+drop table if exists  public.t_demo_text_v1 cascade;   -- customer feedback stored here
+drop table if exists public.t_embeddings_v1 cascade;  -- customer feedback embedded 
+drop table if exists  public.t_ai_report cascade;      -- the ai generated business report
+
+-- make sure pgai docker is running normally
+CREATE EXTENSION IF NOT EXISTS ai CASCADE;
+create extension if not exists vector ;
+
+-- use openai-4o-mini and text-embedding-3-small models for demo purpose. You can use local ollama based on business case
+-- make sure the model connection and availability
+-- temporary set the api_key
+set ai.openai_api_key = 'replace you api key here or use pgai default api_key environment';
+select pg_catalog.current_setting('ai.openai_api_key', true) as api_key;
+
+-- check openai connection
+select * from openai_list_models()  olm
+where id in ('gpt-4o-mini','text-embedding-3-small');
+
+-- check the gpt-4o-mini , text-embedding-3-smal work well.
+with 
+tmp as (select 'what is your name?' as message )
+, test_result as (
+	select
+		message, 
+		openai_chat_complete(
+			'gpt-4o-mini',
+			jsonb_build_array(
+				jsonb_build_object('role', 'user', 'content', message)
+			)) as chat, 
+		openai_embed('text-embedding-3-small',message) as embed
+	from tmp)
+select * from test_result;
+
+
+---- step 2 create customer feedback data table and insert the demo data.
+--- the original customer feedback message to food 
+CREATE TABLE public.t_demo_text_v1 (
+id bigserial NOT NULL,
+title text NOT NULL,
+url text NULL,
+customer_message text NULL,
+text_length INTEGER GENERATED ALWAYS AS (LENGTH(customer_message)) stored,
+CONSTRAINT t_demo_text_v1_pkey PRIMARY KEY (id)
+);
+--- insert data of t_demo_text_v1
+INSERT INTO public.t_demo_text_v1 (title,url,customer_message) VALUES
+	 ('Review pizza',NULL,'The best pizza I''ve ever eaten. The sauce was so tangy!'),
+	 ('Review pizza',NULL,'The pizza was disgusting. I think the pepperoni was made from rats.'),
+	 ('Review pizza',NULL,'I ordered a hot-dog and was given a pizza, but I ate it anyway.'),
+	 ('Review pizza',NULL,'I hate pineapple on pizza. It is a disgrace. Somehow, it worked well on this izza though.'),
+	 ('Review pizza',NULL,'I ate 11 slices and threw up. The pizza was tasty in both directions.');
+
+-- the data of embedding result
+CREATE TABLE public.t_embeddings_v1 (
+id bigserial NOT NULL,
+text_id text NOT NULL, 
+text_content text NOT NULL, -- it is same as t_demo_text_v1
+model_name text NOT NULL,
+ntoken int4 NULL,
+nlength int4 NULL,
+embedding public.vector NOT NULL,
```

Comment:
```suggestion
embedding public.vector(1536) NOT NULL,
```
specify the dimensions

Reply:
Thanks for fix it. I also enabled the create index part which disabled early this no dimension issue.

Reply:
done